lib/ipmi_backend_utils.pm

```diff
@@ -213,6 +213,42 @@ sub get_installation_partition {
     return $partition;
 }
 
+sub set_dom0_mem {
+    my ($root_prefix) = @_;
+    $root_prefix = "/" if (!defined $root_prefix) || ($root_prefix eq "");
+    my $installation_disk = "";
+
+    if ($root_prefix ne "/") {
+        $installation_disk = get_installation_partition;
+        assert_script_run("cd /");
+        mount_installation_disk("$installation_disk", "$root_prefix");
+    }
+
+    assert_script_run('mount --rbind /proc /mnt/proc');
+    assert_script_run('mount --rbind /sys /mnt/sys');
+    assert_script_run('mount --rbind /dev /mnt/dev');
+    type_string("chroot /mnt\n");
+    wait_still_screen;
+
+    # Mount Btrfs sub-volumes
+    assert_script_run('mount -a');
+
+    assert_script_run ". /etc/default/grub";
+    assert_script_run "sed -i '/GRUB_CMDLINE_XEN_DEFAULT/c\\GRUB_CMDLINE_XEN_DEFAULT=\"\$GRUB_CMDLINE_XEN_DEFAULT dom0_mem=4096M\"' /etc/default/grub";
+    assert_script_run "cat /etc/default/grub";
+    assert_script_run "grub2-mkconfig -o /boot/grub2/grub.cfg";
```

Comment:
Does this need to differ for grub1 and grub2?

Reply:
We test SLES12SP1+ and if I'm not mistaken it has already GRUB2.

Reply:
Yes, you are right. SLE12+ only has grub2. If you plan to make this code works for all openqa tests, then you should consider that( We actually have sle11sp4 host tests :)). If not plan to serve all, suggest to add a comment.

Reply:
Thank you :) I don't have any SLES11SP4 working test so I will just comment this limitation.